blas/tpls/KokkosBlas2_gemv_tpl_spec_decl.hpp

```diff
@@ -786,4 +786,105 @@ KOKKOSBLAS2_CGEMV_ROCBLAS(Kokkos::LayoutRight, Kokkos::Experimental::HIPSpace,
+#include <oneapi/mkl/blas.hpp>
+#include <KokkosBlas_tpl_spec.hpp>
+
+namespace KokkosBlas {
+namespace Impl {
+
+#ifdef KOKKOS_ENABLE_SYCL
+
+inline oneapi::mkl::transpose mode_kk_to_onemkl(char mode_kk) {
+  switch (toupper(mode_kk)) {
+    case 'N': return oneapi::mkl::transpose::nontrans;
+    case 'T': return oneapi::mkl::transpose::trans;
+    case 'C': return oneapi::mkl::transpose::conjtrans;
+    default:;
+  }
+  throw std::invalid_argument(
+      "Invalid mode for oneMKL (should be one of N, T, C)");
+}
+
+#define KOKKOSBLAS2_GEMV_ONEMKL(SCALAR, STDSCALAR, LAYOUT, MEM_SPACE,          \
+                                ETI_SPEC_AVAIL)                                \
+  template <>                                                                  \
+  struct GEMV<                                                                 \
+      Kokkos::View<const SCALAR**, LAYOUT,                                     \
+                   Kokkos::Device<Kokkos::Experimental::SYCL, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<const SCALAR*, LAYOUT,                                      \
+                   Kokkos::Device<Kokkos::Experimental::SYCL, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<SCALAR*, LAYOUT,                                            \
+                   Kokkos::Device<Kokkos::Experimental::SYCL, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      true, ETI_SPEC_AVAIL> {                                                  \
+    using execution_space = Kokkos::Experimental::SYCL;                        \
+    using device_type     = Kokkos::Device<execution_space, MEM_SPACE>;        \
+    using mem_traits      = Kokkos::MemoryTraits<Kokkos::Unmanaged>;           \
+    using AViewType =                                                          \
+        Kokkos::View<const SCALAR**, LAYOUT, device_type, mem_traits>;         \
+    using XViewType =                                                          \
+        Kokkos::View<const SCALAR*, LAYOUT, device_type, mem_traits>;          \
+    using YViewType = Kokkos::View<SCALAR*, LAYOUT, device_type, mem_traits>;  \
+                                                                               \
+    static void gemv(const typename AViewType::execution_space& space,         \
+                     const char kk_trans[],                                    \
+                     typename AViewType::const_value_type& alpha,              \
+                     const AViewType& A, const XViewType& X,                   \
+                     typename YViewType::const_value_type& beta,               \
+                     const YViewType& Y) {                                     \
+      execution_space exec{};                                                  \
```

Comment:
If you do this you will force execution on Kokkos' default sycl queue, you want to use the input parameter `space` instead so users can pick the queue if they need to.